File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
# 1<= s <=50
# s: zero시작하는거 x

# z,o,t,f,s,e,n

def solution(s):
    alpa_dict = {'zero':'0','one':'1','two':'2','three':'3','four':'4','five':'5','six':'6','seven':'7','eight':'8','nine':'9'}
    
    num = ['0','1','2','3','4','5','6','7','8','9']
    answer = ''
    tmp = ''
    for i in s:
        tmp += i
        if tmp in alpa_dict :
            answer += alpa_dict.get(tmp)
            tmp = ''
        elif tmp in num: 
            answer += tmp
            tmp = ''
    
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const nums = {
        "zero" : "0",
        "one" : "1",
        "two" : "2",
        "three" : "3",
        "four" :"4",
        "five" : "5",
        "six" : "6",
        "seven" : "7",
        "eight" : "8",
        "nine" : "9"
    }
    
    let number = "", alpha = ""
    s.split("").forEach((val) => {
        isNaN(Number(val)) ? alpha += val : number += val 
        if (alpha in nums) { 
            number += nums[alpha]
            alpha = ""
        }
    })
    return Number(number)
}

// function solution(s) {
//     const num = {
//         "zero" : "0",
//         "one" : "1",
//         "two" : "2",
//         "three" : "3",
//         "four" :"4",
//         "five" : "5",
//         "six" : "6",
//         "seven" : "7",
//         "eight" : "8",
//         "nine" : "9"
//     }
    
//     let result = ""
//     for(let i = 0; i < s.length; i++){
//         if(isNaN(s[i] * 10) === false) {
//             result += s[i]
//         }
//         else{
//             str = ""
//             while(isNaN(s[i] * 10)){
//                 str += s[i];
//                 if (num[str] !== undefined){
//                     result += num[str]
//                     break
//                 }
//                 i++;    
//             }
//         }
//     }
//     return parseInt(result);
// }